### DISCUSSION TITLE
--mouse argument makes laptop touchpad cease to work

### DISCUSSION THREAD

**Discussion by Ascyt** [xremap/xremap#631] [Q&A]
If I run xremap with the `--mouse` argument:

```sh
sudo xremap ./config.yml -- mouse
```

it gives me the output:

```
Selecting devices from the following list:
------------------------------------------------------------------------------
/dev/input/event0 : Power Button
/dev/input/event1 : Lid Switch
/dev/input/event10: HP WMI hotkeys
/dev/input/event11: HD-Audio Generic HDMI/DP,pcm=3
/dev/input/event12: HD-Audio Generic Mic
/dev/input/event13: HD-Audio Generic Headphone
/dev/input/event2 : Video Bus
/dev/input/event3 : AT Translated Set 2 keyboard
/dev/input/event4 : Logitech G502 HERO Gaming Mouse
/dev/input/event5 : Logitech G502 HERO Gaming Mouse Keyboard
/dev/input/event6 : Wireless hotkeys
/dev/input/event7 : PC Speaker
/dev/input/event8 : ELAN071A:00 04F3:30FD Mouse
/dev/input/event9 : ELAN071A:00 04F3:30FD Touchpad
------------------------------------------------------------------------------
Selected keyboards and mice automatically since --device options weren't specified:
------------------------------------------------------------------------------
/dev/input/event3 : AT Translated Set 2 keyboard
/dev/input/event4 : Logitech G502 HERO Gaming Mouse
/dev/input/event5 : Logitech G502 HERO Gaming Mouse Keyboard
/dev/input/event8 : ELAN071A:00 04F3:30FD Mouse
/dev/input/event9 : ELAN071A:00 04F3:30FD Touchpad
------------------------------------------------------------------------------
```

And as you can see here, my laptop's touchpad is detected successfully. 

However, running xremap with this argument `--mouse` makes my touchpad cease to work, I cannot use it to move my mouse cursor anymore until I close xremap. The only thing I can do with my touchpad is left click, I cannot move, right click or anything. External mice work fine.

**Comment by dev-sfuentes** [comment]
Maybe the ELAN071A:00 04F3:30FD Touchpad only has the buttons of the touchpad, have you tried to connect all devices at the same time, to maybe see if the touchpad is also in another /dev/input/event? For me, my headphones are separated in different events and, if I miss one, I lose some functionalities

You could try to debug it by having --mouse --device "" (when used through the cli, it connects to all devices) and see if the touchpad still doesn't work

**Comment by wistoft** [comment]
I can't tell if there's a problem with device selection too, but I have confirmed the problem.

There are events from touchpads, that xremap doesn't handle. First [BTN_TOUCH is filtered out here](https://github.com/xremap/xremap/blob/master/src/device.rs#L48), so it can't be emitted. Second [there isn't added any absolute axes here](https://github.com/xremap/xremap/blob/master/src/device.rs#L62), so those events can't be emitted either.

So touchpads aren't support yet. Only the BTN_LEFT and BTN_RIGHT on touchpads work.
